infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/msghandlers/LogReplicationServer.java

```diff
@@ -138,15 +139,25 @@ public void shutdown() {
      * @return the session for the given request
      */
     private LogReplicationSession getSession(RequestMsg request) {
-        LogReplicationSession session;
+        LogReplicationSession session = null;
 
         if(!request.getHeader().hasSession()) {
-            // Backward compatibility where 'session' field not present
-            session = LogReplicationSession.newBuilder()
-                    .setSourceClusterId(getUUID(request.getHeader().getClusterId()).toString())
-                    .setSinkClusterId(localClusterId)
-                    .setSubscriber(LogReplicationConfigManager.getDefaultSubscriber())
-                    .build();
+            // Backward compatibility where 'session' field is not present.
```

Comment:
nit: It will be useful to mention that the assumption is that the Sink cluster upgrades first followed by the Source